File: testluyentap/app/src/main/java/com/example/testluyentap/MainActivity.java
package com.example.testluyentap;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;

import androidx.activity.EdgeToEdge;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.graphics.Insets;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;

public class MainActivity extends AppCompatActivity {
    Button Btn1,Btn2,Btn3,Btn4;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        EdgeToEdge.enable(this);
        setContentView(R.layout.activity_main);
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main), (v, insets) -> {
            Insets systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars());
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom);
            return insets;

        });
        Btn1= findViewById(R.id.btn1);
        Btn2= findViewById(R.id.btn2);
        Btn3= findViewById(R.id.btn3);
        Btn4= findViewById(R.id.btn4);
        Btn1.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent= new Intent(MainActivity.this,ActivityCau1.class);
                startActivity(intent);
            }
        });
        Btn2.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent= new Intent(MainActivity.this, ActivityCau2.class);
                startActivity(intent);
            }
        });
        Btn3.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent= new Intent(MainActivity.this, ActivityCau3.class);
                startActivity(intent);
            }
        });
        Btn4.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent= new Intent(MainActivity.this, ActivityCau4.class);
                startActivity(intent);
            }
        });
    }
}

File: TinhBMI/app/src/main/java/com/example/tinhbmi/MainActivity.kt
package com.example.tinhbmi

import android.os.Bundle
import android.widget.Button
import android.widget.EditText
import android.widget.TextView
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        enableEdgeToEdge()
        setContentView(R.layout.activity_main)
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main)) { v, insets ->
            val systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars())
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom)
            insets
        }
        val Chieucao = findViewById<EditText>(R.id.input1)
        val Cannang = findViewById<EditText>(R.id.input2)
        val btn = findViewById<Button>(R.id.btn)
        val Kq = findViewById<TextView>(R.id.kq)

        btn.setOnClickListener{
            TinhToan(
                Chieucao, Cannang, Kq,
            )
        }
    }
    private fun TinhToan(Chieucao:EditText,Cannang:EditText,Kq:TextView){
                val a=Chieucao.text.toString().toDouble()
                val b=Cannang.text.toString().toDouble()
                val BMI=b/(a*a)
                val KetQua= when{
                    BMI < 18.5 -> "cân nặng thấp gầy"
                    BMI in 18.5..24.9 -> "bình thường"
                    BMI in 25.0..29.9 ->"tiền béo phì"
                    BMI in 30.0..34.9 ->"béo phì 1"
                    BMI in 35.0..39.9 ->"béo phì 2"
                    BMI >=40.0 ->"béo phì 3"
                    else -> "ko xác định"
                }
            Kq.text="Kết quả là:$KetQua"

    }
}